### PR TITLE
[PORT] Vomitgoose

### PR DESCRIPTION
**Pull request by Cenrus**
https://github.com/tgstation/tgstation/pull/44563

Will be deadchat controllable after the deadchat control component is added, which I will do in a seperate PR

:cl: plapatin, sprites by cogwerks and edited by mrdoombringer
add: vomi/tg/oose
/:cl:

### CODE REVIEW

**Comment by ike709** [comment]
You're missing *a lot* of follow-up fix PRs.

**Comment by Cenrus** [comment]
I only found one, but if there are more I'll commit them.

**Comment by Couls** [comment]
OH NO

**Comment by Cenrus** [comment]
Oh yes

**Comment by ike709** [comment]
Conflicts.

**Comment by ike709** [comment]
https://github.com/tgstation/tgstation/pull/47551/files

**Comment by Cenrus** [comment]
Will get that in a seperate PR.

**Comment by ike709** [comment]
Considering how surprisingly buggy and awful vomitgoose has historically been, this definitely needs a testmerge.

File: code/modules/mob/living/simple_animal/hostile/goose.dm [hunk 1]
```diff
@@ -25,33 +27,151 @@
...
+/mob/living/simple_animal/hostile/retaliate/goose/vomit/attacked_by(obj/item/O, mob/user)
 	. = ..()
-	if(prob(5))
-		Retaliate()
+	feed(O)
 
-/mob/living/simple_animal/hostile/retaliate/goose/proc/update_icon()
-	if(stat != DEAD)
-		icon_state = resting ? icon_resting : icon_living
+/mob/living/simple_animal/hostile/retaliate/goose/vomit/proc/feed(obj/item/reagent_containers/food/tasty)
```
Comment (ike709) [inline code/modules/mob/living/simple_animal/hostile/goose.dm:77]: This proc doesn't check if the object actually is food, causing runtimes when it isn't.

**Comment by ike709** [comment]
Fix the cause of runtimes and I'll merge.

**Pull request merged**